modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/services/StateChannelSnapshotService.scala

```diff
@@ -44,6 +49,20 @@ object StateChannelSnapshotService {
     new StateChannelSnapshotService[F] {
       private val logger = Slf4jLogger.getLogger
 
+      private val sendRetries = 5
+
+      private val retryPolicy: RetryPolicy[F] = RetryPolicies.limitRetries(sendRetries)
+
+      private def wasSuccessful: Either[NonEmptyList[StateChannelValidationError], Unit] => F[Boolean] =
+        _.isRight.pure[F]
+
+      private def onFailure(binaryHashed: Hashed[StateChannelSnapshotBinary]) =
+        (_: Either[NonEmptyList[StateChannelValidationError], Unit], details: RetryDetails) =>
+          logger.info(s"Retrying sending ${binaryHashed.hash.show} to Global L0 after rejection. Retries so far ${details.retriesSoFar}")
+
+      private def onError(binaryHashed: Hashed[StateChannelSnapshotBinary]) = (_: Throwable, details: RetryDetails) =>
```

Comment:
it doesn't need to be private

Reply:
Yes but it's easier to notice it's in this scope only and the IDE's don't complain about no return type signature, that's why I decided to use the private modifier.

Reply:
:+1: works for me